File: pt-web/src/types/article.ts
export type ArticleBlockImage = {
  type: "image";
  src: string;
  alt?: string;
};

export type ArticleBlockText = {
  type: "text";
  content: string;
};

export type ArticleBlockTitle = {
  type: "title";
  content: string;
};

export type ArticleBlockSeparator = {
  type: "separator";
};

export type ArticleBlock =
  | ArticleBlockImage
  | ArticleBlockText
  | ArticleBlockTitle
  | ArticleBlockSeparator;

export type ArticleSummary = {
  id: string;
  slug: string;
  title: string;
  excerpt: string;
  coverUrl: string;
  alt?: string | null;
  author?: string | null;
  featured: boolean;
  createdAt: string;
};

export type Article = {
  id: string;
  slug: string;
  title: string;
  excerpt: string;
  content: string;
  coverUrl: string;
  alt?: string | null;
  author?: string | null;
  featured: boolean;
  createdAt: string;
  blocks?: ArticleBlock[];
};

File: pt-general-go/internal/domain/article.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type ArticleBlock struct {
	Type    string `json:"type"`
	Content string `json:"content,omitempty"`
	Src     string `json:"src,omitempty"`
	Alt     string `json:"alt,omitempty"`
}

type Article struct {
	CreatedAt time.Time
	Alt       *string
	Author    *string
	Slug      string
	Title     string
	Excerpt   string
	Content   string
	CoverURL  string
	ID        uuid.UUID
	Featured  bool
	Blocks    []ArticleBlock
}
